pokemon-checker/src/App.tsx

```diff
@@ -4,20 +4,20 @@ import "./components/PokemonSearch/PokemonSearchResults";
 import PokemonSearchResults from "./components/PokemonSearch/PokemonSearchResults";
 import SearchBox from "./components/SearchBar/SearchBar";
 import PokemonDisplay from "./components/PokemonDisplay/PokemonDisplay";
-import PokemonSearchObj from "./components/PokemonSearch/PokemonSearchResults";
-import { getAllPokemon } from "./services/PokemonService";
+import { PokemonService } from "./services/PokemonService";
+import PokemonDTO from "./DataTransferObjects/PokemonDTO";
 
 type AppState = {
-  pokemonList: typeof PokemonSearchObj[];
+  pokemonList: PokemonDTO[];
   pokemonName: string;
   searchfield: string;
   pokemonSelected: string;
   pokemonUrl: string;
 };
 
 class App extends Component<any, AppState> {
-  private pokemonApiUrl: string =
-    "https://pokeapi.co/api/v2/pokemon?limit=1000offset=0";
+  private pokeService: PokemonService;
```

Comment:
The App now has a proper dependency on the Pokemon Service. This full fetch URL has been moved there.